client/lib/signup/step-actions/index.js

```diff
@@ -740,3 +752,28 @@ export function isSiteTopicFulfilled( stepName, defaultDependencies, nextProps )
 		flows.excludeStep( stepName );
 	}
 }
+
+export function shouldShowUpsell( stepName, defaultDependencies, nextProps ) {
```

Comment:
I always expect functions that start with `should` to be pure and probably to just return a boolean.

Maybe we could try renaming to `maybeShowUpsell()`? The name's a bit awkward, but all I could think of 😅

Reply:
Ah yes, it's used as one of those step fulfilled callbacks. The naming of these isn't very good across the board. Maybe `showOrHideUpsellStep()`?

Reply:
You're right with the naming. I'll push a change with the renamed function.

Reply:
Done in [75596ad](https://github.com/Automattic/wp-calypso/pull/40879/commits/75596ad43d43381391b65db799e083ee0367a740)